eureka2-core/src/main/java/com/netflix/eureka2/registry/SourcedEurekaRegistryImpl.java

```diff
@@ -67,12 +67,21 @@
     private final EurekaRegistryMetrics metrics;
     private final NotifyingInstanceInfoHolder.NotificationTaskInvoker invoker;
 
-    @Inject
     public SourcedEurekaRegistryImpl(EurekaRegistryMetricFactory metricsFactory) {
-        this(metricsFactory, Schedulers.computation());
+        this(metricsFactory, new IndexRegistryImpl<InstanceInfo>(), Schedulers.computation());
     }
 
     public SourcedEurekaRegistryImpl(EurekaRegistryMetricFactory metricsFactory, Scheduler scheduler) {
+        this(metricsFactory, new IndexRegistryImpl<InstanceInfo>(), scheduler);
+    }
+
+    @Inject
+    public SourcedEurekaRegistryImpl(EurekaRegistryMetricFactory metricsFactory, IndexRegistry indexRegistry) {
+        this(metricsFactory, indexRegistry, Schedulers.computation());
+    }
+
+    public SourcedEurekaRegistryImpl(EurekaRegistryMetricFactory metricsFactory, IndexRegistry<InstanceInfo> indexRegistry, Scheduler scheduler) {
```

Comment:
Minor convention, can we have the params be indexRegistry, metricsFactory, Scheduler in that order?